### PR TITLE
Update regions for new sub based on failures

### PR DESCRIPTION
**Pull request by jsturtevant**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

I went through the last 24hrs of activity logs in the new sub and found the following failures:

```
	"The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_D2s_v3' is currently not available in location 'eastus2'. Please try another size or deploy to a different location or different zone. See https://aka.ms/azureskunotavailable for details.\"}]}}",

	The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_B2s' is currently not available in location 'westus2'. 
	The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_B2s' is currently not available in location 'westeurope'.
	The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_D2s_v3' is currently not available in location 'westus2'.
	The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_B2s' is currently not available in location 'eastus'. 
	The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_B2s' is currently not available in location 'eastus2'
```

After reviewing the capacity, these new regions seem to have room (at least don't signal that they are at lower capacity than normal )

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

File: hack/util.sh [hunk 1]
```diff
@@ -63,7 +63,7 @@ capz::util::should_build_ccm() {
 
 # all test regions must support AvailabilityZones
 capz::util::get_random_region() {
-    local REGIONS=("australiaeast" "canadacentral" "eastus" "eastus2" "northcentralus" "northeurope" "uksouth" "westeurope" "westus2")
+    local REGIONS=("australiaeast" "canadacentral" "germanywestcentral" "mexicocentral" "northcentralus" "northeurope" "uksouth" "switzerlandnorth" "brazilsouth")
```
Comment (mboersma) [inline hack/util.sh:66]: `germanywestcentral` and `francecentral` have increased quota and I think would be safe to try, but unless you were able to increase it (I wasn’t), we only have standard quota in `mexicocentral`, `brazilsouth`, and `switzerlandnorth` so we may bump up against that. 
Also could we keep the list alpha-sorted? I just find it easier to do navigate and to diff.

Comment (jsturtevant) [reply hack/util.sh:66]: what is "standard quota"?  I was able to bump the quotas for D2v3 and B2s past 150 for all them (most more than that)

**Comment by jsturtevant** [comment]
AKS jobs don't seem to work in due to Capacity Restrictions: `Standard_D2s_v3' is currently not available in location 'brazilsouth'`

Am wondering if we should be using the newer version sku's? Maybe they just starting remove the v3 version from datacenters? It seems every region we choose has issues....

File: hack/util.sh (hunk 1 above)
Comment (mboersma) [reply hack/util.sh:66]: "Standard quota" or default appears to be 65. If you were able to raise it, then full steam ahead. I had tried increasing quota in those regions previously but been denied.

**Comment by nojnhuh** [comment]
I don't see any downsides to trying newer SKUs. Are there any we have quota for?

**Comment by jsturtevant** [comment]
these look equivalent https://learn.microsoft.com/en-us/azure/virtual-machines/ddv5-ddsv5-series#ddsv5-series and so far I have been able to request in quota increases in the regions listed here

**Comment by jsturtevant** [comment]
mexicocentral doesn't support storage v2 apparently, removed

File: templates/test/ci/prow-aks-aso/patches/aks-pool2.yaml [hunk 2]
```diff
@@ -35,5 +35,5 @@ spec:
         name: ${CLUSTER_NAME}
       mode: User
       type: VirtualMachineScaleSets
-      vmSize: "${AZURE_AKS_NODE_MACHINE_TYPE:=Standard_D2s_v3}"
+      vmSize: "${AZURE_AKS_NODE_MACHINE_TYPE:=Standard_D2ds_v5}"
```
Comment (nojnhuh) [inline templates/test/ci/prow-aks-aso/patches/aks-pool2.yaml:38]: Is there also a newer version of Standard_B2s we could add to the self-managed templates if we're still seeing capacity issues for those too?

**Comment by nojnhuh** [comment]
Looking for capacity issue flakes:
/test pull-cluster-api-provider-azure-e2e-aks

**Comment by nojnhuh** [comment]
One more for good measure. Three in a row is enough to convince me this is definitely an improvement.
/test pull-cluster-api-provider-azure-e2e-aks

**Comment by nojnhuh** [comment]
This is the error I see on the latest AKS run:
```
      UltraSSD is not available for VM size Standard_D2ds_v5 in availability zone 2 of region switzerlandnorth.: PUT https://management.azure.com/subscriptions/===REDACTED===/resourceGroups/capz-e2e-v2lrok-aks/providers/Microsoft.ContainerService/managedClusters/capz-e2e-v2lrok-aks
      --------------------------------------------------------------------------------
      RESPONSE 400: 400 Bad Request
      ERROR CODE: ErrorCodeEnableUltraSSD
      --------------------------------------------------------------------------------
      {
        "code": "ErrorCodeEnableUltraSSD",
        "details": null,
        "message": "UltraSSD is not available for VM size Standard_D2ds_v5 in availability zone 2 of region switzerlandnorth.",
        "subcode": ""
      }
      --------------------------------------------------------------------------------
```

Is there a way we can figure out which other regions might have the same issue without running the tests?

**Comment by jsturtevant** [comment]
Are we requesting UltraSSD?  

> Is there a way we can figure out which other regions might have the same issue without running the tests?

There must a be better way...

**Comment by jsturtevant** [comment]
The other tests failures don't see to be related to sku's so I think in general the change was in the right direction.

**Comment by nojnhuh** [comment]
Yes, we are defining UltraSSD:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/89efbf34282c373765819e5c3c4a7e3bff2d20de/templates/test/ci/cluster-template-prow-aks.yaml#L83

I see we don't do the same with the ASOAPI template, and I wonder if that's why it has been pretty stable despite the rest of the tests hitting lots of capacity errors: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-pr-e2e-aks-main

Do we have a separate quota for UltraSSD disks that we might need to bump?

**Comment by nojnhuh** [comment]
If it's not an UltraSSD thing, it might be an availability zones thing?

**Comment by jsturtevant** [comment]
do we need ` enableUltraSSD: true ` set? are we testing that for a reason?

**Comment by nojnhuh** [comment]
> do we need `enableUltraSSD: true` set? are we testing that for a reason?

I don't think I was around when that was first added, so I'm not sure. I'm thinking that's not the issue though since I can see self-managed tests failing to create VMs without setting the (I'm assuming) equivalent `spec.additionalCapabilities.enableUltraSSD` field on AzureMachines.

**Comment by nojnhuh** [comment]
I'm hoping #5034 mostly fixes the capacity issues for the AKS tests.

**Comment by nojnhuh** [comment]
These are the regions where there are currently no zonal restrictions on Standard_B2s:
```
australiaeast     
canadacentral     
centralindia      
eastasia          
francecentral     
germanywestcentral
israelcentral     
italynorth        
koreacentral      
mexicocentral     
northeurope       
norwayeast        
polandcentral     
southafricanorth  
spaincentral      
swedencentral     
switzerlandnorth  
uksouth           
```

I'll check to see which of those we have quota for.

**Comment by nojnhuh** [comment]
These are the ones with more than what looks like the default quota:
```
northeurope
australiaeast
canadacentral
uksouth
francecentral
germanywestcentral
switzerlandnorth
```

**Comment by nojnhuh** [comment]
> This is the error I see on the latest AKS run:
> 
> ```
>       UltraSSD is not available for VM size Standard_D2ds_v5 in availability zone 2 of region switzerlandnorth.: PUT https://management.azure.com/subscriptions/===REDACTED===/resourceGroups/capz-e2e-v2lrok-aks/providers/Microsoft.ContainerService/managedClusters/capz-e2e-v2lrok-aks
>       --------------------------------------------------------------------------------
>       RESPONSE 400: 400 Bad Request
>       ERROR CODE: ErrorCodeEnableUltraSSD
>       --------------------------------------------------------------------------------
>       {
>         "code": "ErrorCodeEnableUltraSSD",
>         "details": null,
>         "message": "UltraSSD is not available for VM size Standard_D2ds_v5 in availability zone 2 of region switzerlandnorth.",
>         "subcode": ""
>       }
>       --------------------------------------------------------------------------------
> ```


Updated the regions and removed UltraSSD from the AKS test templates to avoid this error.

File: hack/util.sh [hunk 3]
```diff
@@ -63,7 +63,7 @@ capz::util::should_build_ccm() {
 
 # all test regions must support AvailabilityZones
 capz::util::get_random_region() {
-    local REGIONS=("australiaeast" "canadacentral" "eastus" "eastus2" "northcentralus" "northeurope" "uksouth" "westeurope" "westus2")
+    local REGIONS=("australiaeast" "canadacentral" "francecentral" "germanywestcentral" "northeurope" "switzerlandnorth" "uksouth" )
```
Comment (nojnhuh) [inline hack/util.sh:66]: francecentral had a restriction on the v5 SKU but not the one we currently use.

File: templates/test/ci/prow-aks/patches/aks-pool0.yaml [hunk 4]
```diff
@@ -7,7 +7,6 @@ spec:
   osDiskType: "Managed"
   osDiskSizeGB: 30
   enableNodePublicIP: false
-  enableUltraSSD: true
```
Comment (willie-yao) [inline templates/test/ci/prow-aks/patches/aks-pool0.yaml:10]: Looks good. Did you check if ultra SSDs are available in the new regions?

**Review by willie-yao — APPROVED** [review]
/lgtm

File: templates/test/ci/prow-aks/patches/aks-pool0.yaml (hunk 4 above)
Comment (nojnhuh) [reply templates/test/ci/prow-aks/patches/aks-pool0.yaml:10]: They evidently are not in switzerlandnorth, but this removes all usage of UltraSSD in our e2e tests so we shouldn't be hitting that kind of error anymore.

**Pull request merged**